sentry/src/main/java/io/sentry/config/provider/PropertiesConfigurationProvider.java

```diff
@@ -0,0 +1,28 @@
+package io.sentry.config.provider;
+
+import java.util.Properties;
+
+import io.sentry.util.Nullable;
+
+/**
+ * A trivial wrapper to use a {@link Properties} instance as a configuration provider.
+ */
+public class PropertiesConfigurationProvider implements ConfigurationProvider {
```

Comment:
I suggest we could live without this type.
One less abstraction cost of calling `properties == null ? null : properties.getProperty(key);` in all implementations.

If I'm not missing anything, this far we only have 1 implementation anyway.